File: contribution/aquarium/features/logging.rb
require 'aquarium'
require 'company_logger'

module Logging
  include Aquarium::Aspects
  
  @logger = CompanyLogger.new

  def self.logger
    @logger
  end
  
  call_cut = JoinPoint.new :type => Employee, :method => :cut, :on_types => [Employee]
  write_salary = JoinPoint.new :type => /^Employee/, :method => :salary=, :on_types => [Employee]
  
  Aspect.new :around, :pointcuts => [call_cut, write_salary] do |join_point, emp, *args|
    old_salary = emp.salary
    join_point.proceed
    p @logger.log(emp.name,old_salary,emp.salary)
  end
end

File: contribution/springAOP/src/main/java/org/softlang/aspects/Logging.java
package org.softlang.aspects;

import org.aspectj.lang.ProceedingJoinPoint;
import org.aspectj.lang.annotation.Around;
import org.aspectj.lang.annotation.Aspect;
import org.softlang.company.Employee;
import org.softlang.utils.CompanyLogger;
import org.springframework.stereotype.Component;

@Aspect
@Component
public class Logging {

	private CompanyLogger logger = CompanyLogger.getInstance();

	@Around("(execution(public * org.softlang.services.EmployeeService.cut(org.softlang.company.Employee)) "
			+ "|| execution(public * org.softlang.services.EmployeeService.setEmployeeSalary(..))) "
			+ "&& args(emp,..)")
	public void beforeCallAnyMethod(ProceedingJoinPoint pjp, Employee emp) throws Throwable {
		double oldSalary = emp.getSalary();
		pjp.proceed();
		logger.logCut(emp.getName(), oldSalary, emp.getSalary());
		
	}

	public CompanyLogger getLogger() {
		return logger;
	}

}
